### PR TITLE
Modular cmake

### PR DESCRIPTION
**Pull request by AntoinePrv**
Reopening #334 & added default value for `OPTION_PREFIX` based on `CMAKE_PROJECT_NAME`.

### CODE REVIEW

**Comment by AntoinePrv** [comment]
@JohanMabille I tried this locally with Xwidgets, and I can simply do
```cmake
find_package(xeus  REQUIRED)
include("${xeus_DIR}/CompilerWarnings.cmake")

xeus_target_add_compile_warnings(mytarget)
```
Which is quite nice. We can certainly improve the interface (among other things) but I think this is a good start.

The warning chosen here are quite aggressive but I don't think we should settle for less.

**Comment by AntoinePrv** [comment]
Actually this is general enough that it could live with `xtl` (or `xcl` :wink: ) so that we can reuse such elements in xtl/xtensor/xproperty, even mamba.

**Comment by AntoinePrv** [comment]
Actually this is general enough that it could live with `xtl` (or `xcl` :wink: ) so that we can reuse such elements in xtl/xtensor/xproperty, even mamba.

File: CMakeLists.txt [hunk 1]
```diff
@@ -216,11 +220,11 @@ macro(xeus_create_target target_name linkage output_name)
     # =================
 
     target_compile_features(${target_name} PRIVATE cxx_std_14)
+    xeus_target_add_compile_warnings(${target_name})
 
     if (CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR
         CMAKE_CXX_COMPILER_ID MATCHES "GNU" OR
         CMAKE_CXX_COMPILER_ID MATCHES "Intel")
-        target_compile_options(${target_name} PUBLIC -Wunused-parameter -Wextra -Wreorder)
```
Comment (JohanMabille) [inline CMakeLists.txt:223]: I think we're missing `extra` and `reorder` in the cmake warning file.

**Review by JohanMabille — CHANGES_REQUESTED** [review]

File: cmake/CompilerWarnings.cmake [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        # applied
+        /w14928
+    )
+
+    set(
+        clang_warnings
+        # Some default set of warnings
+        -Wall
+        # Reasonable and standard
+        -Wextra
```
Comment (AntoinePrv) [inline cmake/CompilerWarnings.cmake:79]: @JohanMabille extra is here.

File: CMakeLists.txt (hunk 1 above)
Comment (AntoinePrv) [reply CMakeLists.txt:223]: `extra` already there. I could not find a `reorder` warning on GCC doc.

Comment (JohanMabille) [reply CMakeLists.txt:223]: Althoug there is no a detailed documentation about it, we can find it in the list of warnings enabled by `-Wall`: https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html

**Review by JohanMabille — APPROVED** [review]

**Pull request merged**